src/api.rs

```diff
@@ -421,6 +422,7 @@ pub struct Events {
 
 impl Events {
```

Comment:
I'd like to add [those tests](https://gist.github.com/zopu) to the crate, with @zopu's permission.

Reply:
@zopu will add them after this is merged :)